### ISSUE TITLE
IDNs in the zone table

### ISSUE THREAD

**Issue by huwsteadCL** [NICMx/rdap-server#103]
In regards to the zone table should IDNs be added as punycode (i.e. xn--)?

We have a number of IDN tlds and when we try and retrieve from a domain query (in which the extension is provided as punycode)  we get a "Zone not found"

Also as a side question I assume that we shouldn't us using punycode conversion when adding in a domain name to the domain.dom_unicode_name field?

**Comment by dhfelix** [comment]
> In regards to the zone table should IDNs be added as punycode (i.e. xn--)?
> We have a number of IDN tlds and when we try and retrieve from a domain query (in which the extension is provided as punycode) we get a "Zone not found"

The code needs a fix to support IDN in zones. 

> Also as a side question I assume that we shouldn't us using punycode conversion when adding in a domain name to the domain.dom_unicode_name field?

When the code asks for a domain in the database, the domain query information is sent as u-label, so the domain information in the database must be in u-label.

**Comment by huwsteadCL** [comment]
> When the code asks for a domain in the database, the domain query information is sent as u-label, so the domain information in the database must be in u-label.

But I assume the query itself from a URL standpoint can be the Punycode variant?

**Comment by dhfelix** [comment]
yes, the user can ask as a-label or u-label, in fact the server module just check if it is a valid domain name, our rdap-sql-provider searches in the db in u-label only.

**Comment by dhfelix** [comment]
I just release a new version of [RDAP-core.jar](https://github.com/NICMx/rdap-core/releases/download/1.2.1/rdap-core-1.2.1.jar)  and [Rdap-sql-provider.jar](https://github.com/NICMx/rdap-sql-provider/releases/download/v1.4.2/rdap-sql-provider-1.4.2.jar) , replace your old Jars with the new ones, and thats it, it should work.

I still need to update the documentation, and maybe update the rdap-server.war with this new jars

**Comment by huwsteadCL** [comment]
Thanks @dhfelix 

In regards to the zone.zone_name should be zone name be stored as a-label or u-label in the database?

**Comment by dhfelix** [comment]
Oh right, my bad sorry, It should be stored in u-label format.
I think it will have the same problem as the issue #106, let me check in our test environments.
